SystemSettings.php

```diff
@@ -240,4 +244,79 @@ private function settingToCountryCodes(Setting $setting)
         }
         return $codes;
     }
+	
+	private function createIpRangeAllowlistSetting()
+	{
+		return $this->makeSetting('iprange_allowlist', [], FieldConfig::TYPE_ARRAY, function (FieldConfig $field) {
+			$field->title = Piwik::translate('TrackingSpamPrevention_IpAllowlistTitle');
+			$field->description = Piwik::translate('TrackingSpamPrevention_IpAllowlistDescription');
+			$field->uiControl = FieldConfig::UI_CONTROL_MULTI_TUPLE;
+
+			$field1 = new FieldConfig\MultiPair('CIDR', 'cidr', FieldConfig::UI_CONTROL_TEXT);
+			$field->uiControlAttributes['field1'] = $field1->toArray();
+
+			$self = $this;
+			$field->transform = function ($value) use ($self) {
+				return $self->transformCidrsList($value);
+			};
+
+			$field->validate = function ($rows) {
+				if (empty($rows) || !is_array($rows)) {
+					return;
+				}
+				foreach ($rows as $row) {
+					$cidr = is_array($row) ? (string) ($row['cidr'] ?? '') : (string) $row;
+					$cidr = trim($cidr);
+					if ($cidr === '') {
+						continue;
+					}
+
+					if (strpos($cidr, '/') === false) {
+						if (IPUtils::stringToBinaryIP($cidr) === false) {
+							throw new \Exception("Invalid IP or CIDR: {$cidr}");
```

Comment:
Can you move all this exceptions text in a translation file.
Ideally you should add the text with an appropriate key name in `lang/en.json` and use that 
```suggestion
							throw new \Exception(Piwik::translate('TrackingSpamPrevention_IvalidIPOrCidrExceptionMessage'));
```